### PR TITLE
Added a check for Attributes under 0 and set to Zero

### PR DESCRIPTION
**Pull request by jessica38**
To Fix Bug #96 

We Added a check for any value under Zero and then set that key to Zero.

### CODE REVIEW

**Comment by LJNIC** [comment]
If we're going to loop through every attribute an additional time, you may as well move that line into the `getAllAttributes` call so you only have to loop a single additional time. I'd also advise replacing the conditional with `Math.max(0.0, value)` instead. Finally, remember to keep style in mind (run the default IntelliJ Code -> Reformat to be sure).

File: src/main/java/com/atherys/rpg/service/AttributeService.java [hunk 1]
```diff
@@ -129,6 +129,8 @@ public AttributeService() {
      */
     public Map<AttributeType, Double> mergeAttributes(Map<AttributeType, Double> source, Map<AttributeType, Double> additional) {
         additional.forEach((type, value) -> source.merge(type, value, Double::sum));
+        //Added Check for Stats under 0; Replace with 0;
+        additional.forEach((k,v) -> { if (v <= 0){additional.replace(k,v,0.0); }});
```
Comment (AaronFoley) [inline src/main/java/com/atherys/rpg/service/AttributeService.java:133]: ```additional``` is returned but probably not used, this function modifies the ```source``` map, we really should be returning source here I think?

The check most likely needs to be done in the first forEach loop that modifies the ```source``` map

**Comment by jessica38** [comment]
Hey, umm yea we can move it to there. @LJNIC can you just read over math.max and ensure that there's a reason for a calculation on zero. I'm sorry. 

@AaronFoley after reviewing myself again I think we should do the loop before we merge with source. And I'm assuming that source isn't gonna be negative

**Comment by LJNIC** [comment]
Using `Math.max` is just simpler. I also noticed you could replace it with just `replaceAll`!

**Comment by jessica38** [comment]
Replace all is a replacement for all the values 

This is great if we are changing the whole map 

I'm guessing you would do something like?

If (math.max(0, value){
//set key to a zero value
}

That's fine I will make the adjustment tomorrow afternoon

**Comment by LJNIC** [comment]
I still think you should move it down to the main function and use `replaceAll`. That way, we only iterate through the attributes once, and it's also more concise. (`replaceAll` is identical to your `forEach`)

**Comment by AaronFoley** [comment]
I still think this won't work?
The important place to remove negative values is in the source map, we are merging the attribute map first (which can result in negative values in the Source map) and then removing negative values from the attribute map, which I'm pretty sure isn't re-used, even tho we return it (we really should be returning the players updated attributes, aka the source map)

**Comment by LJNIC** [comment]
Yeah, just setting them to zero *after* they've all been combined is what I suggested.

**Comment by jessica38** [comment]
Why not create a new function to remove her values from the source map that way we can call it anywhere?

**Comment by AaronFoley** [comment]
Looks good

**Review by AaronFoley — APPROVED** [review]

**Comment by jessica38** [comment]
Just did a bit of reformatting based on Rynelfs Suggestions

**Pull request merged**